### PR TITLE
Update launch template to enable capacity reservation and EFA network interface

### PR DESCRIPTION
**Pull request by alyssa1303**
This pull request introduces significant enhancements to the AWS EKS Terraform module, primarily focused on making launch template configuration more flexible and supporting advanced EC2 features. The changes add support for capacity reservations, spot instance market options, and more granular network interface settings, while also expanding the module's input variable schema and updating test data for new edge cases.

**Launch Template Configuration Enhancements:**

* Added support for dynamic `capacity_reservation_specification` and `instance_market_options` blocks in the `aws_launch_template` resource, allowing specification of capacity reservations and spot instance options directly in node group configuration.
* Refactored `network_interfaces` to be a dynamic block, enabling configuration of properties like `associate_public_ip_address`, `delete_on_termination`, and `interface_type` per node group.

**Input Variable Schema Updates:**

* Extended the `eks_config` variable object to include optional fields for `capacity_reservation_specification`, `instance_market_options`, and `network_interfaces`, enabling these advanced EC2 options to be set via module inputs.
* Added a new top-level variable `capacity_reservation_id` for global reservation configuration.

**Node Group Resource Logic:**

* Updated `instance_types` assignment in `aws_eks_node_group` to handle scenarios where capacity blocks are used, ensuring correct instance type selection.

**Test Data and Edge Case Coverage:**

* Added new mock data and override resources in `data_launch_template.tfmock.hcl` and `data_launch_template_edge_cases.tfmock.hcl` to test various launch template configurations, including edge cases for capacity reservation, spot options, and network interface settings. [[1]](diffhunk://#diff-12b4cff290ab84d4bebede9bc350439d98b4779627e68e932bcae95f0c2137e0R1-R111) [[2]](diffhunk://#diff-7e34813abef0131571fd3a041b6120052afe9999fcda20188a04c28d14c91b2aR1-R111)
* Adjusted test assertions to reflect changes in network interface configuration logic.

**Module Wiring and Locals:**

* Updated `main.tf` to wire new local and variable values (`capacity_reservation_id`) into the EKS module, ensuring end-to-end support for the new features. [[1]](diffhunk://#diff-4fa7e1512b9b19b6cf0ec7166bff84df309a63560e72e26b07a28e5f3fc0a086R8) [[2]](diffhunk://#diff-4fa7e1512b9b19b6cf0ec7166bff84df309a63560e72e26b07a28e5f3fc0a086R62)

### CODE REVIEW

**Review by anson627 — APPROVED** [review]

File: modules/terraform/aws/main.tf [hunk 1]
```diff
@@ -1,10 +1,11 @@
...
-  creation_time    = var.json_input["creation_time"]
-  k8s_machine_type = lookup(var.json_input, "k8s_machine_type", null)
-  ena_express      = lookup(var.json_input, "ena_express", null)
+  region                  = lookup(var.json_input, "region", "us-east-1")
+  run_id                  = lookup(var.json_input, "run_id", "123456")
+  user_data_path          = lookup(var.json_input, "user_data_path", "")
+  creation_time           = var.json_input["creation_time"]
+  k8s_machine_type        = lookup(var.json_input, "k8s_machine_type", null)
+  ena_express             = lookup(var.json_input, "ena_express", null)
+  capacity_reservation_id = lookup(var.json_input, "capacity_reservation_id", null)
```
Comment (sumanthreddy29) [inline modules/terraform/aws/main.tf:8]: Can you add this new variable to aws_input_schema.json file?
